tests/pyfunc/test_inferred_data_path.py

```diff
@@ -0,0 +1,123 @@
+import pickle
+import mlflow
+import os
+from mlflow.models import Model
+import numpy as np
+import pytest
+import sklearn.neighbors
+import sklearn.datasets
+import sys
+
+sys.path.insert(0, ".")
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
+
+
+@pytest.fixture(scope="module")
+def iris_data():
+    iris = sklearn.datasets.load_iris()
+    x = iris.data[:, :2]
+    y = iris.target
+    return x, y
+
+
+@pytest.fixture(scope="module")
+def sklearn_knn_model(iris_data):
+    x, y = iris_data
+    knn_model = sklearn.neighbors.KNeighborsClassifier()
+    knn_model.fit(x, y)
+    return knn_model
+
+
+def _walk_dir(path):
+    for root, dirs, files in os.walk(path):
+        for file in files:
+            yield os.path.relpath(os.path.join(root, file), start=path)
+
+
+def test_loader_module_model_save_load(sklearn_knn_model, iris_data, tmp_path, model_path):
+    sk_model_path = os.path.join(tmp_path, "knn.pkl")
+    with open(sk_model_path, "wb") as f:
+        pickle.dump(sklearn_knn_model, f)
+
+    model_config = Model(run_id="test", artifact_path="testtest")
+    mlflow.pyfunc.save_model(
+        path=model_path,
+        data_path=sk_model_path,
+        loader_module="custom_model.loader",
+        mlflow_model=model_config,
+        infer_code_paths=True,
+    )
+
+    reloaded_model_config = Model.load(os.path.join(model_path, "MLmodel"))
+
+    assert set(_walk_dir(os.path.join(model_path, "code"))) == {
+        'custom_model/loader.py',
+        'custom_model/mod1/__init__.py',
+        'custom_model/mod1/mod2/__init__.py',
+        'custom_model/mod1/mod4.py'
```

Comment:
Can we update mod4.py with an import of some pypi package that would otherwise be included in the base pip requirements to illustrate to others looking at this test that we do not expect that import to be included due to it being external to the repo references?

Reply:
Sure. Good idea in testing.